zoo/policies/__init__.py

```diff
@@ -100,3 +104,86 @@ def human_keyboard_entrypoint(*arg, **kwargs):
 
 
 register(locator="human-in-the-loop-v0", entry_point=human_keyboard_entrypoint)
+
+
+from smarts.env.multi_scenario_env import resolve_agent_interface
+
+
+def load_config(path):
+    import yaml
+
+    config = None
+    if path.exists():
+        with open(path, "r") as file:
+            config = yaml.safe_load(file)
+    return config
+
+
+def competition_entry(**kwargs):
+    policy_path = kwargs.get("policy_path", None)
+
+    from .competition_agent import CompetitionAgent
+
+    def env_wrapper(env):
+        import gym
+
+        # insert policy path
+        if policy_path in sys.path:
+            sys.path.remove(policy_path)
+        
+        sys.path.insert(0, policy_path)
+
+        # import policy.py module
+        wrapper_path = str(os.path.join(policy_path, "policy.py"))
+        wrapper_spec = importlib.util.spec_from_file_location(
+            "competition_wrapper", wrapper_path
+        )
+        wrapper_module = importlib.util.module_from_spec(wrapper_spec)
+        sys.modules["competition_wrapper"] = wrapper_module
+        if wrapper_spec:
+            wrapper_spec.loader.exec_module(wrapper_module)
+
+        wrappers = wrapper_module.submitted_wrappers()
+        
+        # delete competition wrapper module and remove path
+        sys.modules.pop("competition_wrapper")
+        del wrapper_module
+        sys.path.remove(policy_path)
+
+        env = gym.Wrapper(env)
+        for wrapper in wrappers:
+            env = wrapper(env)
+
+        return env
+
+    config = load_config(Path(os.path.join(policy_path, "config.yaml")))
+
+    spec = AgentSpec(
+        interface=resolve_agent_interface(
+            img_meters=int(config["img_meters"]),
+            img_pixels=int(config["img_pixels"]),
+            action_space="TargetPose",
+        ),
+        agent_params={
+            "policy_path": policy_path,
+        },
+        adapt_env=env_wrapper,
+        agent_builder=CompetitionAgent,
+    )
+
+    return spec
+
+
+root_path = str(Path(__file__).absolute().parents[2])
+
+register(
+    "competition_agent-v0",
+    entry_point=competition_entry,
+    policy_path=os.path.join(root_path, "competition/track1/submission_1"),
+)
+
+register(
+    "competition_agent-v1",
```

Comment:
I wonder if these should have distinct names, and then have `v0` for both.

Reply:
The competition_agent-v1 will be removed in the end after testing some competition policies.